pkg/image/image.go

```diff
@@ -214,31 +223,38 @@ func (i *Image) Read() error {
 	i.Layers = layers
 
 	// in order to resolve symlinks all squashed trees must be available
-	return i.squash(readProg)
+	err = i.squash(readProg)
```

Comment:
Note: I want to come back to this on a second pass:

Should `i.SquashedTreee()` call `i.squash()`?

Does squash() affect `fileCatalog`?

Coming back to look here in a second

Reply:
> Should i.SquashedTreee() call i.squash()?

no -- squash is an expensive operation, we do this on Read() only and used the cached squashed trees for later usage.

> Does squash() affect fileCatalog?

no it should not -- the squash only makes structural changes / additions, but does not create any new files, so the catalog doesn't get modified.